internal/cssc/cssc.go

```diff
@@ -0,0 +1,190 @@
+// Reads the filter json from the specified file path and returns the Filter struct
+func GetFilterFromFilePath(filePath string) (Filter, error) {
+	file, err := os.ReadFile(filePath)
+	if err != nil {
+		return Filter{}, errors.Wrap(err, "error reading the filter json file from file path")
+	}
+
+	var filter = Filter{}
+	if err := json.Unmarshal(file, &filter); err != nil {
+		return Filter{}, errors.Wrap(err, "error unmarshalling json content when reading the filter file from file path")
+	}
+
+	return filter, nil
+}
+
+// Applies filter to filter out the repositories and tags from the ACR according to the specified criteria and returns the FilteredRepository struct
+func ApplyFilterAndGetFilteredList(ctx context.Context, acrClient api.AcrCLIClientInterface, filter Filter) ([]FilteredRepository, error) {
+	var filteredRepos []FilteredRepository
+
+	for _, filterRepo := range filter.Repositories {
+		if filterRepo.Enabled != nil && !*filterRepo.Enabled {
+			continue
+		}
+		if filterRepo.Repository == "" || filterRepo.Tags == nil || len(filterRepo.Tags) == 0 || filterRepo.Tags[0] == "" {
+			continue
+		}
+
+		tagList, err := tag.ListTags(ctx, acrClient, filterRepo.Repository)
+		if err != nil {
+			if strings.Contains(err.Error(), "404") {
+				continue
+			}
+			return nil, err
+		}
+
+		if len(filterRepo.Tags) == 1 && filterRepo.Tags[0] == "*" {
+			for _, tag := range tagList {
+				if strings.Contains(*tag.Name, "-patched") {
+					originalTag := strings.Split(*tag.Name, "-patched")[0]
+					matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: originalTag, PatchTag: *tag.Name}
+					filteredRepos = AppendElement(filteredRepos, matchingRepo)
+				} else {
+					matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: *tag.Name, PatchTag: *tag.Name}
+					filteredRepos = AppendElement(filteredRepos, matchingRepo)
+				}
+			}
+		} else {
+			for _, ftag := range filterRepo.Tags {
+				for _, tag := range tagList {
+					if *tag.Name == ftag {
+						matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: *tag.Name, PatchTag: *tag.Name}
+						filteredRepos = AppendElement(filteredRepos, matchingRepo)
+					} else if *tag.Name == ftag+"-patched" {
+						matchingRepo := FilteredRepository{Repository: filterRepo.Repository, Tag: ftag, PatchTag: ftag + "-patched"}
+						filteredRepos = AppendElement(filteredRepos, matchingRepo)
+					}
+				}
+			}
+		}
+	}
+	return filteredRepos, nil
+}
+
+// Prints the filtered result to the console
+func PrintFilteredResult(filteredResult []FilteredRepository, showPatchTags bool, loginURL string) {
+	if len(filteredResult) == 0 {
+		fmt.Println("No matching repository and tag found!")
+	}
+	if showPatchTags {
+		fmt.Println("Listing repositories and tags matching the filter with corrosponding patch tag (if present):")
+		for _, result := range filteredResult {
+			fmt.Printf("%s/%s:%s,%s\n", loginURL, result.Repository, result.Tag, result.PatchTag)
```

Comment:
Is there a need to include the loginurl for each of these?

Reply:
yeah, I've done that to keep the output consistent with other pre-existing commands like acr tag list